cparser.lua

```diff
@@ -25,8 +25,58 @@ local function enum_handler(cursor, parent, declarations)
         table.insert(declarations, decl)
 end
 
-local function function_handler(cursor, parent, declarations)
+local function union_struct_handler(cursor, declarations) 
+        local fields = {}
+        local definition = {}
```

Comment:
I would rename it to be consistent with the current vocabulary that we are using. e.g. Clang calls struct definitions StructDecl.
```suggestion
        local chield_declarations = {}
```
Another option would be inserting all declarations in the top-level. This option might be better even though we lose a bit of info.